### DISCUSSION TITLE
GitHub Copilot is too expensive

### DISCUSSION THREAD

**Discussion by iXingo** [community/community#30364] [Copilot Conversations]
I love GitHub Copilot, but it is too expensive for me

**Comment by melvincwng** [comment]
@iXingo , do you happen to be a student? If you are, GitHub Copilot is free for students! Just fyi 🙂

**Reply by hexxone** [reply]
You know, all the public Source Code on Github which Copilot was trained with is FREE aswell- but Microsoft just decided to monetize on it. Making it free for Students does nothing.

It's the same as if you were to compose an Original piece of music, upload it to YouTube- but someone claims it, starts playing ADs on it, takes the revenue and gives you no Option of appeal.

**Reply by iXingo** [reply]
> You know, all the public Source Code on Github which Copilot was trained with is FREE aswell- but Microsoft just decided to monetize on it. Making it free for Students does nothing.
> 
> It's the same as if you were to compose an Original piece of music, upload it to YouTube- but someone claims it, starts playing ADs on it, takes the revenue and gives you no Option of appeal.

I agree with you

**Reply by bgolinvaux** [reply]
Do not underestimate the price of the computing resources that analyze and make code predictions (with a very short low latency). Furthermore, there are certainly hundreds of engineering man.hours behind it.

I have found that, at least for Python, Copilot often saves me a lot of keystrokes. But I agree that the price should be dependent, let's say, on the average local developer income.

**Reply by iXingo** [reply]
Yes, if they want to charge for the computing resources and the engineering cost, Github Copilot should specify different prices by country. You know $10 is expensive for an engineer like me in China.

**Reply by chiefmikey** [reply]
> Yes, if they want to charge for the computing resources and the engineering cost, Github Copilot should specify different prices by country. You know $10 is expensive for an engineer like me in China.

I think this is a good point and should really apply to any virtual services. Obviously resources need to be considered but it would be simple enough to adjust the price for differences in currency inflation to keep the cost to value consistent for everyone in the global community. If $10 for you isn't what $10 is for me but we both pay $10 for Copilot, the real cost is different but the value or service we are both receiving doesn't change.

**Reply by minesunny** [reply]
> > Yes, if they want to charge for the computing resources and the engineering cost, Github Copilot should specify different prices by country. You know $10 is expensive for an engineer like me in China.
> 
> I think this is a good point and should really apply to any virtual services. Obviously resources need to be considered but it would be simple enough to adjust the price for differences in currency inflation to keep the cost to value consistent for everyone in the global community. If $10 for you isn't what $10 is for me but we both pay $10 for Copilot, the real cost is different but the value or service we are both receiving doesn't change.

世界上最搞笑的事情就是，用发展中国家的资源生产产品，用发达国家的消费水平定义价格；

**Reply by Cyrene963** [reply]
佬你是华为的啊？

**Comment by shpkng** [comment]
The annual payment for Copilot is more expensive than that for a JetBrains IDE, which is not fair IMO. This plugin is great, but not worth this money.

**Reply by reorx** [reply]
not to mention it uses the code we create

**Comment by bartleby** [comment]
You're lucky! Guys from Russia can't use it at all :(

**Reply by minesunny** [reply]
中国人在抱怨它很贵，俄罗斯人在说他们用不了

**Comment by quantuminformation** [comment]
greedy pricing by m$oft

**Comment by zubair2001** [comment]
you can get it for free if you are a student

**Comment by foamzou** [comment]
I am the maintainer of open source project. Could github grant a free copilot to me? Thanks

**Comment by rishavbhowmik** [comment]
How I understand their pricing is.

To be fair, GitHub Copilot has a lot of execution overheads, which has to be paid for. That easily justifies 4$/month, for unlimited usage.

Since GitHub Copilot is the only auto-complete tool of it's kind (in terms of usability, precision, performance etc).
MS will charge for the exclusiveness of the product, until there is any competition.
So that adds up another 3$/month

Thus 7$/month.

**Reply by cobaltautomationdev** [reply]
10rmb as apple music in china, I can accept it.   $10 US dollars is about 70rmb per month, it is too expensive.

**Comment by arohman84** [comment]
the price should based on country, this $10 cheap on US but for indonesian like me, its so expensive

**Comment by leandrojo** [comment]
1USD = 5BRL 😢

**Reply by ronaldosjr** [reply]
550BRL (almost half brazilian minimum wage) per year is way too expensive for most brazilians

**Reply by rotnet** [reply]
I agree with you

**Comment by sinanisler** [comment]
10 dolar = 188 TL

**Reply by sinanisler** [reply]
update 10 dolar = 290 TL :)

**Reply by lucas-labs** [reply]
I get you. I'm from Argentina, we share the torture of having probably the two worst currencies in the world haha.
10 dollars = 10000 argentine pesos.... mine is wort 😭

**Reply by lucas-labs** [reply]
Good news here, 10 dollars are still ~10000 argentine pesos almost 2 years later. First time in decades I can say this. Hooray! 😅

**Comment by simanga-dev** [comment]
I am from South Africa, I love the tool but I just can't afford it

**Comment by sinanisler** [comment]
10 USD = 280 TL

there should be regional pricing like Netflix, prime, steam games, and microsoft games........
this is stupid.

**Comment by chewenkai** [comment]
Can some official guys reply this thread? Thanks.

**Comment by abdullahhafizh** [comment]
I hope the price didnt increase after some new features got released

**Comment by ogatawikan** [comment]
The product is very helpful, but unfortunately it is too expensive for us in Indonesia, especially those who work outside metropolitan cities, 10 USD can be used for 1 week's food

**Comment by axa88** [comment]
Same issue in Vietnam. i suggested many of our young programmers get co pilot to assist them, especially since documentation is often only available in English which not everyone understands well. 

 But at current conversion rates, 10$ = 250,000d, typically they will purchase lunch on the street for about 30k, so thats 1 month of copilot or almost 2 weeks of prepared lunch. Which would you choose?

MS wouldn't be losing money with regional pricing. no one here is purchasing it anyway.

**Reply by ITPC-Khmer** [reply]
too expensive for me
